rust/src/smb/smb.rs

```diff
@@ -53,6 +53,82 @@ use crate::smb::events::*;
 use crate::smb::files::*;
 use crate::smb::smb2_ioctl::*;
 
+#[repr(C)]
+pub enum SMBFrameType {
+    NBSS = 0,
+    NBSSHdr = 1,
+    NBSSData = 2,
+    SMB1 = 3,
+    SMB1Hdr = 4,
+    SMB1Data = 5,
+    SMB2 = 6,
+    SMB2Hdr = 7,
+    SMB2Data = 8,
+    SMB3 = 9,
+    SMB3Hdr = 10,
+    SMB3Data = 11,
+}
+
+impl SMBFrameType {
+    fn from_u8(value: u8) -> Option<SMBFrameType> {
+        match value {
+            0 =>  Some(SMBFrameType::NBSS),
+            1 =>  Some(SMBFrameType::NBSSHdr),
+            2 =>  Some(SMBFrameType::NBSSData),
+            3 =>  Some(SMBFrameType::SMB1),
+            4 =>  Some(SMBFrameType::SMB1Hdr),
+            5 =>  Some(SMBFrameType::SMB1Data),
+            6 =>  Some(SMBFrameType::SMB2),
+            7 =>  Some(SMBFrameType::SMB2Hdr),
+            8 =>  Some(SMBFrameType::SMB2Data),
+            9 =>  Some(SMBFrameType::SMB3),
+            10 => Some(SMBFrameType::SMB3Hdr),
+            11 => Some(SMBFrameType::SMB3Data),
+            _ => None,
+        }
+    }
+}
+
+fn smb_frame_type_string(s: &str) -> i32 {
```

Comment:
Can we use a single datatype to represent the enum as an integer?  u8?  Looking at the derive macro here.

Reply:
storing this as `uint8_t` on the C side.